lib/circuitbox/circuit_breaker.rb

```diff
@@ -52,15 +52,17 @@ def run!(run_options = {})
         logger.debug "[CIRCUIT] closed: querying #{service}"
 
         begin
-          t1 = Time.now
+          start_time = Time.now.to_i
           response = if exceptions.include? Timeout::Error
             timeout_seconds = run_options.fetch(:timeout_seconds) { option_value(:timeout_seconds) }
             timeout (timeout_seconds) { yield }
           else
             yield
           end
-          t2 = Time.now
-          log_circuit_execution_time(t2 - t1)
+          end_time = Time.now.to_i
+          if circuit_options[:notify_circuit_execution_time]
```

Comment:
The only thing I see us running into with that is when you want different time measurements you'll have to create a new class for each one since this class is initialized when the circuit is initialized.

Say for instance I wanted to use the monotonic clock in milliseconds for some work that was longer running and microseconds for one that was more latency sensitive. I might have something along the lines of
```ruby
class MonotonicTime
  def initialize(notifier, metric_name)
    @notifier = notifier
    @metric_name = metric_name
    @time_unit = :milliseconds
  end

 def run
    # pretty much the same logic here
  end
end

class MonotonicTimeMicroseconds < MonotonicTime
  def initialize(notifier, metric_name)
    super
    @time_unit = :microsecond
  end
end
```

I think we might be able to alleviate my concerns by doing something like this

```ruby
class Timed
  def run(notifier, metric_name)
    before = Time.now
    yield
    after = Time.now
    notifier.metric_gauge(metric_name, before - after)
  end
end
```
and then change the options fetching to
``` @time_class   = options.fetch(:timer) { Timed.new } ```
We could also change run to be a class method and then the timer can just be ```Timed```

This makes it possible for us to create a monotonic timer like this
```ruby
class MonotonicTime
  def initialize(time_unit)
    @time_unit = time_unit
  end

  def run(notifier, metric_name)
    before = Process.clock_gettime(Process::CLOCK_MONOTONIC, @time_unit)
    yield
    after = Process.clock_gettime(Process::CLOCK_MONOTONIC, @time_unit)
    notifier.metric_gauge(metric_name, before - after)
  end
end
```
and that can be initialized by the user when creating the circuit allowing them to specify the time unit.

thoughts?